### ISSUE TITLE
Open MPI process cannot find a corresponding process entry for that peer.

### ISSUE THREAD

**Issue by egorgam** [horovod/horovod#2477]
I have two workers and one master node. When I run `horovodrun -np 1 -H localhost:1 python3 train.py` on each node, or `mpirun --allow-run-as-root --mca oob_tcp_if_include eth0 --mca btl_tcp_if_include eth0 -np 2 -x NCCL_SOCKET_IFNAME=eth0  -np 1 -H worker1:1 -bind-to none -map-by slot -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH -mca pml ob1 -mca btl ^openib python3 train.py` (or with worker2) all works good. (train.py is tensporfow_keras synthetic benchmark). But when I try to start both workers from master I get this: 

```
root@ubuntu:~/hrv# mpirun --allow-run-as-root --mca oob_tcp_if_include eth0 --mca btl_tcp_if_include eth0 -np 2 -x NCCL_SOCKET_IFNAME=eth0  -np 2 -H worker1:1,worker2:1 -bind-to none -map-by slot -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH -mca pml ob1 -mca btl ^openib python3 train.py
2020-11-24 03:59:21.002354: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2020-11-24 03:59:20.986180: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
--------------------------------------------------------------------------
WARNING: Open MPI accepted a TCP connection from what appears to be a
another Open MPI process but cannot find a corresponding process
entry for that peer.

This attempted connection will be ignored; your MPI job may or may not
continue properly.

  Local host: ee7a3819e74b
  PID:        231539
```

and nothing work do.

**Comment by egorgam** [comment]
May this problem be if workers has same internal IP on eth0?

**Comment by egorgam** [comment]
log with --mca btl_base_verbose 100:
```root@ubuntu:~/hrv# mpirun --allow-run-as-root --mca oob_tcp_if_include eth0 --mca btl_base_verbose 100 --mca btl_tcp_if_include eth0 -np 2 -x NCCL_SOCKET_IFNAME=eth0  -np 2 -H worker1:1,worker2:1 -bind-to none -map-by slot -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH -mca pml ob1 -mca btl ^openib python3 train.py
2020-11-24 04:07:30.802106: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2020-11-24 04:07:30.803181: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
[ee7a3819e74b:231568] mca: base: components_register: registering framework btl components
[ee7a3819e74b:231568] mca: base: components_register: found loaded component sm
[ee7a3819e74b:231568] mca: base: components_register: found loaded component tcp
[ee7a3819e74b:231568] mca: base: components_register: component tcp register function successful
[ee7a3819e74b:231568] mca: base: components_register: found loaded component vader
[ee7a3819e74b:231568] mca: base: components_register: component vader register function successful
[ee7a3819e74b:231568] mca: base: components_register: found loaded component self
[ee7a3819e74b:231568] mca: base: components_register: component self register function successful
[ee7a3819e74b:231568] mca: base: components_open: opening btl components
[ee7a3819e74b:231568] mca: base: components_open: found loaded component tcp
[ee7a3819e74b:231568] mca: base: components_open: component tcp open function successful
[ee7a3819e74b:231568] mca: base: components_open: found loaded component vader
[ee7a3819e74b:231568] mca: base: components_open: component vader open function successful
[ee7a3819e74b:231568] mca: base: components_open: found loaded component self
[ee7a3819e74b:231568] mca: base: components_open: component self open function successful
[ee7a3819e74b:231568] select: initializing btl component tcp
[ee7a3819e74b:231568] btl:tcp: Attempting to bind to AF_INET port 1024
[ee7a3819e74b:231568] btl:tcp: Successfully bound to AF_INET port 1024
[ee7a3819e74b:231568] btl:tcp: my listening v4 socket is 0.0.0.0:1024
[ee7a3819e74b:231568] btl:tcp: examining interface eth0
[ee7a3819e74b:231568] btl:tcp: using ipv6 interface eth0
[c4809972e6a0:186950] mca: base: components_register: registering framework btl components
[c4809972e6a0:186950] mca: base: components_register: found loaded component sm
[c4809972e6a0:186950] mca: base: components_register: found loaded component tcp
[c4809972e6a0:186950] mca: base: components_register: component tcp register function successful
[c4809972e6a0:186950] mca: base: components_register: found loaded component vader
[c4809972e6a0:186950] mca: base: components_register: component vader register function successful
[c4809972e6a0:186950] mca: base: components_register: found loaded component self
[c4809972e6a0:186950] mca: base: components_register: component self register function successful
[c4809972e6a0:186950] mca: base: components_open: opening btl components
[c4809972e6a0:186950] mca: base: components_open: found loaded component tcp
[c4809972e6a0:186950] mca: base: components_open: component tcp open function successful
[c4809972e6a0:186950] mca: base: components_open: found loaded component vader
[c4809972e6a0:186950] mca: base: components_open: component vader open function successful
[c4809972e6a0:186950] mca: base: components_open: found loaded component self
[c4809972e6a0:186950] mca: base: components_open: component self open function successful
[c4809972e6a0:186950] select: initializing btl component tcp
[c4809972e6a0:186950] btl:tcp: Attempting to bind to AF_INET port 1024
[c4809972e6a0:186950] btl:tcp: Successfully bound to AF_INET port 1024
[c4809972e6a0:186950] btl:tcp: my listening v4 socket is 0.0.0.0:1024
[c4809972e6a0:186950] btl:tcp: examining interface eth0
[c4809972e6a0:186950] btl:tcp: using ipv6 interface eth0
[c4809972e6a0:186950] select: init of component tcp returned success
[c4809972e6a0:186950] select: initializing btl component vader
[c4809972e6a0:186950] select: init of component vader returned failure
[c4809972e6a0:186950] mca: base: close: component vader closed
[c4809972e6a0:186950] mca: base: close: unloading component vader
[c4809972e6a0:186950] select: initializing btl component self
[c4809972e6a0:186950] select: init of component self returned success
[ee7a3819e74b:231568] select: init of component tcp returned success
[ee7a3819e74b:231568] select: initializing btl component vader
[ee7a3819e74b:231568] select: init of component vader returned failure
[ee7a3819e74b:231568] mca: base: close: component vader closed
[ee7a3819e74b:231568] mca: base: close: unloading component vader
[ee7a3819e74b:231568] select: initializing btl component self
[ee7a3819e74b:231568] select: init of component self returned success
[c4809972e6a0:186950] mca: bml: Using self btl for send to [[39949,1],0] on node c4809972e6a0
[ee7a3819e74b:231568] mca: bml: Using self btl for send to [[39949,1],1] on node ee7a3819e74b
[ee7a3819e74b:231568] btl:tcp: path from 172.28.0.2 to 172.28.0.2: IPV4 PRIVATE SAME NETWORK
[ee7a3819e74b:231568] mca: bml: Using tcp btl for send to [[39949,1],0] on node worker1
[ee7a3819e74b:231568] btl: tcp: attempting to connect() to [[39949,1],0] address 172.28.0.2 on port 1024
[ee7a3819e74b:231568] btl:tcp: would block, so allowing background progress
[ee7a3819e74b:231568] btl:tcp: connect() to 172.28.0.2:1024 completed (complete_connect), sending connect ACK
--------------------------------------------------------------------------
WARNING: Open MPI accepted a TCP connection from what appears to be a
another Open MPI process but cannot find a corresponding process
entry for that peer.

This attempted connection will be ignored; your MPI job may or may not
continue properly.

  Local host: ee7a3819e74b
  PID:        231568
-------------------------```

**Comment by tgaddair** [comment]
Hey @egorgam, what happens when you try running `mpirun` from master and one remote host, or from one remote host to the other?  Is there any difference in behavior?

Also, could you share the results of running `ifconfig` on the two worker hosts?

**Comment by egorgam** [comment]
Yes I have same behavior when I tried `mpirun` from other host in my configuration. And I sure a problem exists because all of my workers have same 172.28.0.2 address in their eth0 interface. In my case I can't change workers IP addresses or create new interface. But I wish maybe it possible to use hosts with same IP adresses (and not same hostnames) with mpirun

**Comment by tgaddair** [comment]
Hey @egorgam, yes, in your case I believe doing something similar to the approach described [here](https://github.com/horovod/horovod/blob/master/docs/docker.rst#running-containers-with-different-ports) should work.

**Comment by egorgam** [comment]
@tgaddair thanks for answer, I am already using ssh aliases but it looks like btl is working with workers internal IP, and some collision occurred after ACK to first worker:

[c4809972e6a0:186950] mca: bml: Using self btl for send to [[39949,1],0] on node c4809972e6a0
[ee7a3819e74b:231568] mca: bml: Using self btl for send to [[39949,1],1] on node ee7a3819e74b
[ee7a3819e74b:231568] btl:tcp: path from 172.28.0.2 to 172.28.0.2: IPV4 PRIVATE SAME NETWORK
[ee7a3819e74b:231568] mca: bml: Using tcp btl for send to [[39949,1],0] on node worker1
[ee7a3819e74b:231568] btl: tcp: attempting to connect() to [[39949,1],0] address 172.28.0.2 on port 1024
[ee7a3819e74b:231568] btl:tcp: would block, so allowing background progress
[ee7a3819e74b:231568] btl:tcp: connect() to 172.28.0.2:1024 completed (complete_connect), sending connect ACK
--------------------------------------------------------------------------

**Comment by tgaddair** [comment]
Hey @egorgam, can you try running with `horovodrun --gloo` instead of MPI?  It may be that this issue is specific to MPI.

**Comment by egorgam** [comment]
if I use `--gloo` with `horovodrun` I got this:

```
[1]<stderr>:2020-11-25 22:32:14.710913: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
[0]<stderr>:2020-11-25 22:32:14.728265: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
[0]<stderr>:terminate called after throwing an instance of 'gloo::IoException'
[0]<stderr>:  what():  [/tmp/pip-install-ghn4px9_/horovod/third_party/compatible_gloo/gloo/transport/tcp/pair.cc:769] connect [172.28.0.2]:11173: Connection refused
[0]<stderr>:bash: line 1: 16279 Aborted                 (core dumped) HOROVOD_HOSTNAME=worker1 HOROVOD_RANK=0 HOROVOD_SIZE=2 HOROVOD_LOCAL_RANK=0 HOROVOD_LOCAL_SIZE=1 HOROVOD_CROSS_RANK=0 HOROVOD_CROSS_SIZE=2 SHELL=/bin/bash PWD=/root LOGNAME=root XDG_SESSION_TYPE=tty MOTD_SHOWN=pam HOME=/root LANG=C.UTF-8 LS_COLORS='rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:' LC_TERMINAL=iTerm2 SSH_CONNECTION='109.252.85.218 1630 95.217.1.172 22' LESSCLOSE='/usr/bin/lesspipe %s %s' XDG_SESSION_CLASS=user TERM=xterm-256color LESSOPEN='| /usr/bin/lesspipe %s' USER=root LC_TERMINAL_VERSION=3.4.2 SHLVL=1 XDG_SESSION_ID=336 LC_CTYPE=C.UTF-8 XDG_RUNTIME_DIR=/run/user/0 SSH_CLIENT='109.252.85.218 1630 22' PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/0/bus SSH_TTY=/dev/pts/4 _=/usr/local/bin/horovodrun HOROVOD_STALL_CHECK_TIME_SECONDS=60 HOROVOD_STALL_SHUTDOWN_TIME_SECONDS=0 HOROVOD_NUM_NCCL_STREAMS=1 HOROVOD_CCL_BGT_AFFINITY=0 HOROVOD_GLOO_TIMEOUT_SECONDS=30 PYTHONUNBUFFERED=1 HOROVOD_GLOO_RENDEZVOUS_ADDR=95.217.1.172 HOROVOD_GLOO_RENDEZVOUS_PORT=19678 HOROVOD_CONTROLLER=gloo HOROVOD_CPU_OPERATIONS=gloo HOROVOD_GLOO_IFACE=eth0 NCCL_SOCKET_IFNAME=eth0 python3 train.py
Process 0 exit with status code 134.
```

**Comment by egorgam** [comment]
so how I understand, most important message is` connect [172.28.0.2]:11173: Connection refused`. 172.28.0.2 is IP address of all my wokers in them eth0 interface. When I try to use --gloo with only one worker, everything works.

Funny, but one time I runned both workers successfully - when I reinstalled horovod with `HOROVOD_WITH_GLOO=1 HOROVOD_WITHOUT_MPI=1` and was run mpirun on master, but job was duplicated, workers didn't understood they was in cluster

**Comment by egorgam** [comment]
I found a commentary for my mpirun error
https://github.com/open-mpi/ompi/blob/master/opal/mca/btl/tcp/help-mpi-btl-tcp.txt#L116

**Comment by feacluster** [comment]
I too ran into this bizarre error, "WARNING: Open MPI accepted a TCP connection from what appears to be another Open MPI process but cannot find a corresponding process".

After spending days of googling and lots of back and forth with the vendor, I found the root cause. From:

https://www.mail-archive.com/users@lists.open-mpi.org/msg34182.html

That typically occurs when some nodes have multiple interfaces, and
several nodes have a similar IP on a private/unused interface.

So there was a virbr0 network on captainmarvel03 and captaimarvel01 with the same IP address. Per:

https://forums.centos.org/viewtopic.php?t=61634

I disabled and stopped the libvirtd service and rebooted. Now everything works like before. The problem happened after I had installed GNOME desktop on captainmarvel03..
